src/main.js

```diff
@@ -195,6 +195,21 @@ module.exports = {
     this.subscriptions.dispose()
   },
 
+  async restartESLintWorker() {
+    return new Promise((resolve) => {
+      if (this.worker !== null) {
+        this.worker.terminate()
+        this.worker = null
+      }
+      const initializeESLintWorker = () => {
```

Comment:
Since this code is in multiple places it should be moved to a separate function of its own.